factorio-map-process.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+set -euo pipefail
+## Defaults
+SERVER_BASE_PATH=/srv/factorio/maps
+MAP_TILES_PATH=images
+FACTORIO_MAP_SCRIPT="$(dirname "${BASH_SOURCE[0]}")/factoriomap.py"
+TARFILE=
+REDUCE=0
+
+function usage() {
+    cat <<EOF
+USAGE
+
+factorio-map-process.sh [OPTIONS] tarfile
+
+DESCRIPTION
+
+This script will process a given tarfile into a set of images used in a leaflet map.
+The default behavior will create a directory for the world maps if one does not exist and copy the example html page in as 'index.html'
+The tar file will be parsed for a map name and a date, separated by underscores (_). Files should be of the format {WorldName}_YYYY-MM-DD.tar.
+    
+OPTIONS
+
+    -t | --tiles <images>           Set the name of the directory to store the tile images in.  Default value is
+                                    'images'
+
+    -s | --server <server path>     Set the base path where the http server files will be stored.  Default value is
+                                    '/srv/factorio/maps'
+
+    -r                              Flag to size-reduce the images folder with the rdfind tool
+EOF
+}
+
+function testRequirements() {
+	if ! [ -x "$(command -v jq)" ]; then
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+        if [[ REDUCE -eq 1 && !(-x "$(command -v rdfind)") ]]; then
+		echo "ERROR: Requirement not satisfied - rdfind -"
+		exit 1
+	fi
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: ${FILENAME}"
+
+	IFS='\n'
+	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< "${NAME_NOEXT}"))
+	unset IFS
```

Comment:
I recently learned that setting then unsetting `$IFS` isn't a good idea, since you could be unsetting it from somewhere else (maybe even the users shell if they source the script for some reason. Better to use a local:
```suggestion
	local IFS='\n'
	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< "${NAME_NOEXT}"))
```

Reply:
Neat, good to know.